doc/examples/mnist_mlp/dist_mnist_mlp.cpp

```diff
@@ -107,9 +107,9 @@ float test_accuracy(MNistDataLoader& loader,
            static_cast<float>(sample_count);
 }
 
-int main(int argc, const char* argv[])
+int main(int argc, char* argv[])
 {
-    MPI::Init();
+    MLSL::Environment::GetEnv().Init(&argc, &argv);
```

Comment:
This looks like a classic case where RAII would be a better solution
https://en.cppreference.com/w/cpp/language/raii

Reply:
Good idea, thank you